umpost/um2netcdf.py

```diff
@@ -161,6 +168,7 @@ def fix_level_coord(cube, z_rho, z_theta):
                 c_sigma.var_name = 'sigma_theta'
 
 
+# TODO: split cube ops into functions, this will likely increase process() workflow steps
```

Comment:
What's your current idea for this? I'd recommend either:

- Keep `cubewrite` as a wrapper function, and break it up internally; or,
- Split it into 'data modification' and 'actual disk write' parts, and break those up internally as required.

I'd advise against showing too much of the 'sausage making' inside `process()` itself, if it can be avoided.

Reply:
The outline for retro-fixing/testing `cubewrite()` is in line with `process()` rework:
* Extract its half dozen modification code blocks to separate functions & test
* Replace the code blocks with calls to the new functions
* Iteratively reduce `cubewrite()` to a unit testable size & add tests
* Refactor: cube modification calls into `process()`
  - Aim: refactor `cubewrite()` to a function that only takes cubes & writes to NetCDF
  - Gather cube modification steps into the `process()` workflow (improve proximity of modification funcs)
* `process()` is likely to become somewhat oversized
  - Refactor to extract sensible sub-workflows where needed (e.g. cube renaming, time repairs)
  - Gather these under "middle manager" sub-workflow functions & call these from `process()`

Ideally, `process()` will lay out the high level workflow. Mid layer functions will perform larger subtasks & the lowest level is task modification functions. The code currently doesn't have an obvious hierarchy, so it's less readable.

I'm assuming future ESM versions will be released, possibly requiring new or modified workflows. With a modular set of composition friendly components, new/adapted workflows can be built alongside existing ones.